### PR TITLE
Add publish by tag workflow, add publish Dockerfile

### PR DESCRIPTION
**Pull request by haorenfsa**
Signed-off-by: shaoyue.chen <shaoyue.chen@zilliz.com>

### CODE REVIEW

**Comment by haorenfsa** [comment]
@zwd1208  

secrets.DOCKERHUB_USER & secrets.DOCKERHUB_TOKEN need to be added to repository setting.

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    name: Publish Builder
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu18.04]
+    env:
+      OS_NAME: ${{ matrix.os }}
+      REGISTRY_NAME: registry.zilliz.com/milvus/milvus-operator
```
Comment (zwd1208) [inline .github/workflows/docker-image.yml:20]: We need to publish to Dockerhub, so the variable REGISTRY_NAME needs to be changed

**Comment by haorenfsa** [comment]
test result see: 
https://github.com/haorenfsa/milvus-operator/runs/4150443948?check_suite_focus=true
https://hub.docker.com/repository/docker/haorenfsa/milvus-operator

**Pull request merged**